server/src/main/java/io/unitycatalog/server/persist/dao/StorageCredentialDAO.java

```diff
@@ -0,0 +1,128 @@
+package io.unitycatalog.server.persist.dao;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.unitycatalog.server.model.AwsIamRoleResponse;
+import io.unitycatalog.server.model.AzureServicePrincipal;
+import io.unitycatalog.server.model.StorageCredentialInfo;
+import jakarta.persistence.*;
+import java.time.Instant;
+import java.util.Date;
+import java.util.UUID;
+import lombok.*;
+import lombok.experimental.SuperBuilder;
+
+@Entity
+@Table(name = "uc_storage_credential")
+// Lombok
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+@SuperBuilder
+@EqualsAndHashCode(callSuper = true)
+public class StorageCredentialDAO extends IdentifiableDAO {
+  public static ObjectMapper objectMapper = new ObjectMapper();
+
+  @Column(name = "credential_type", nullable = false)
+  @Enumerated(EnumType.STRING)
+  private CredentialType credentialType;
+
+  @Lob
+  @Column(name = "credential", nullable = false)
+  private String credential;
+
+  @Column(name = "comment")
+  private String comment;
+
+  @Column(name = "owner")
+  private String owner;
+
+  @Column(name = "created_at", nullable = false)
+  private Date createdAt;
+
+  @Column(name = "created_by")
+  private String createdBy;
+
+  @Column(name = "updated_at")
+  private Date updatedAt;
+
+  @Column(name = "updated_by")
+  private String updatedBy;
+
+  public static StorageCredentialDAO from(StorageCredentialInfo storageCredentialInfo) {
+    StorageCredentialDAOBuilder storageCredentialDAOBuilder =
+        StorageCredentialDAO.builder()
+            .id(
+                storageCredentialInfo.getId() != null
+                    ? UUID.fromString(storageCredentialInfo.getId())
+                    : null)
+            .name(storageCredentialInfo.getName())
+            .comment(storageCredentialInfo.getComment())
+            .owner(storageCredentialInfo.getOwner())
+            .createdAt(
+                storageCredentialInfo.getCreatedAt() != null
+                    ? Date.from(Instant.ofEpochMilli(storageCredentialInfo.getCreatedAt()))
+                    : new Date())
+            .createdBy(storageCredentialInfo.getCreatedBy())
+            .updatedAt(
+                storageCredentialInfo.getUpdatedAt() != null
+                    ? Date.from(Instant.ofEpochMilli(storageCredentialInfo.getUpdatedAt()))
+                    : null)
```

Comment:
why is this null? if we are updating the record via DAO, shouldnt we always update this and this should never be set to null in the record?